hwy/ops/arm_neon-inl.h

```diff
@@ -6484,6 +6484,406 @@ HWY_API VFromD<D> SlideDownLanes(D d, VFromD<D> v, size_t amt) {
+         class DW = Rebind<MakeWide<TFromD<D>>, D>>
+HWY_API VFromD<DW> WidenMulAdd(D /* tag */, VFromD<D> mul,
+                               VFromD<D> x, VFromD<DW> add) {
+  return VFromD<DW>(vmlal_s32(add.raw, mul.raw, x.raw));
+}
+
+template<class D, HWY_IF_I32(TFromD<D>), HWY_IF_LANES_D(D, 1),
+         class DW = Rebind<MakeWide<TFromD<D>>, D>>
+HWY_API VFromD<DW> WidenMulAdd(D d, VFromD<D> mul,
+                               VFromD<D> x, VFromD<DW> add) {
+  Vec128<int64_t> mulRs = Vec128<int64_t>(vmull_s32(mul.raw, x.raw));
+  const RepartitionToWide<decltype(d)> d64;
+  const VFromD<DW> mul10(LowerHalf(d64, mulRs));
+  return add + mul10;
+}
+
+template<class D, HWY_IF_U32(TFromD<D>), HWY_IF_LANES_D(D, 2),
+         class DW = Rebind<MakeWide<TFromD<D>>, D>>
+HWY_API VFromD<DW> WidenMulAdd(D /* tag */, VFromD<D> mul,
+                               VFromD<D> x, VFromD<DW> add) {
+  return VFromD<DW>(vmlal_u32(add.raw, mul.raw, x.raw));
+}
+
+template<class D, HWY_IF_U32(TFromD<D>), HWY_IF_LANES_D(D, 1),
+         class DW = Rebind<MakeWide<TFromD<D>>, D>>
+HWY_API VFromD<DW> WidenMulAdd(D /* tag */, VFromD<D> mul,
+                               VFromD<D> x, VFromD<DW> add) {
+  Vec128<uint64_t> mulRs = Vec128<uint64_t>(vmull_u32(mul.raw, x.raw));
+  const VFromD<DW> mul10(LowerHalf(mulRs));
+  return add + mul10;
+}
+
+#if HWY_NEON_HAVE_F16C
+template<class D, HWY_IF_F16(TFromD<D>), HWY_IF_LANES_GT_D(D, 4),
+    class DW = RepartitionToWide<D>>
+HWY_API VFromD<DW> WidenLowMulAdd(D /* tag */, VFromD<D> mul,
+                                  VFromD<D> x, VFromD<DW> add) {
+  return VFromD<DW>(vfmlalq_low_f16(add.raw, mul.raw, x.raw));
+}
+
+template<class D, HWY_IF_F16(TFromD<D>),
+    HWY_IF_LANES_LE_D(D, 4), HWY_IF_LANES_GT_D(D, 2),
+    class DW = RepartitionToWide<D>>
+HWY_API VFromD<DW> WidenLowMulAdd(D /* tag */, VFromD<D> mul,
+                                  VFromD<D> x, VFromD<DW> add) {
+  return Vec64<float32_t>(vfmlal_low_f16(add.raw, mul.raw, x.raw));
+}
+
+template<class D, HWY_IF_F16(TFromD<D>), HWY_IF_LANES_D(D, 2),
+    class DW = RepartitionToWide<D>>
+HWY_API VFromD<DW> WidenLowMulAdd(D /* tag */, VFromD<D> mul,
+                                  VFromD<D> x, VFromD<DW> add) {
+  const DW dw;
+  return MulAdd(add, PromoteLowerTo(dw, mul), PromoteLowerTo(dw, x));
+}
+#endif
+
+}
+
+// ------------------------------ WidenMulAccumulate
+
+#ifdef HWY_NATIVE_WIDEN_MUL_ACCUMULATE
+#undef HWY_NATIVE_WIDEN_MUL_ACCUMULATE
+#else
+#define HWY_NATIVE_WIDEN_MUL_ACCUMULATE
+#endif
+
+template<class D, HWY_IF_INTEGER(TFromD<D>), HWY_IF_LANES_GT_D(D, 1),
+    class DW = RepartitionToWide<D>, typename VW = VFromD<DW>>
+HWY_API VW WidenMulAccumulate(D d, VFromD<D> mul, VFromD<D> x, VW low, VW& high) {
+  const Half<decltype(d)> dh;
+  high = detail::WidenHighMulAdd(d, mul, x, high);
+  return detail::WidenMulAdd(dh, LowerHalf(dh, mul), LowerHalf(dh, x), low);
+}
+
+#if HWY_NEON_HAVE_F16C
```

Comment:
This one is only for the f16 conversion instruction. For full support of f16 operations, we check `HWY_HAVE_FLOAT16`.